File: app/src/main/java/com/benoitquenaudon/tvfoot/red/app/injection/component/ScreenComponent.java
package com.benoitquenaudon.tvfoot.red.app.injection.component;

import dagger.Subcomponent;
import com.benoitquenaudon.tvfoot.red.app.injection.module.ActivityModule;
import com.benoitquenaudon.tvfoot.red.app.injection.scope.ScreenScope;

@ScreenScope @Subcomponent public interface ScreenComponent {
  ActivityComponent plus(ActivityModule activityModule);
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/injection/component/ScreenComponent.kt
package com.benoitquenaudon.tvfoot.red.injection.component

import com.benoitquenaudon.tvfoot.red.injection.module.ActivityModule
import com.benoitquenaudon.tvfoot.red.injection.scope.ScreenScope
import dagger.Subcomponent

@ScreenScope @Subcomponent interface ScreenComponent {
  fun plus(activityModule: ActivityModule): ActivityComponent
}
